account-integrations/webauthn/src/lib/FCL_elliptic.sol

```diff
@@ -834,6 +834,40 @@ library FCL_Elliptic_ZZ {
         return x1 == 0;
     }
 
+
+    /**
+     * @dev ECDSA verification, given , signature, and public key.
+     * @notice Modified from "ecdsa_verify" to use memory instead of calldata
+     */
+    function ecdsa_verify_memory(
```

Comment:
Take extra care reviewing this change as I had to replace some inline assembly

Reply:
I'm also not too familiar with EVM opcodes but your logic/explanation makes sense to me.  I also like your test in the EVM playground.

Reply:
Why is the type uint256 instead of casting a bytes32?
The `sub` assembly function probably doesn't care for the type (just the number of bits), but still curious why have `rs0` a different type to the result of `calldataload`?

Also, what's the reasoning behind the change from `calldata` to `memory`?

Reply:
> Why is the type uint256 instead of casting a bytes32?

I was just sticking to the original type. Would you recommend casting to `bytes32`?

> what's the reasoning behind the change from calldata to memory?

The full explanation for why I changed from `calldata` to `memory` is in the PR description. Does it make sense or should I expand upon it?

**TLDR** is that I couldn't find a way to decode the signature without having the variables stored as `memory` because `abi.decode` appears to return `memory` vars. Then without changing FCL, I get the following error when passing these memory vars into the library functions. So I had to change subsequent functions where the vars are used too: 
```
Invalid type for argument in function call. Invalid implicit conversion from bytes memory to bytes calldata requested.
```

I added a snippet in the description which you can try out in remix to test my assumptions. Definitely open to alternatives if anyone knows of any, as I didn't want to make the change